database.go

```diff
@@ -405,14 +405,14 @@ func (m *Measurement) walkWhereForSeriesIds(expr influxql.Expr, filters map[uint
 	switch n := expr.(type) {
 	case *influxql.BinaryExpr:
 		// if it's EQ then it's either a field expression or against a tag. we can return this
-		if n.Op == influxql.EQ || n.Op == influxql.LT || n.Op == influxql.LTE || n.Op == influxql.GT ||
-			n.Op == influxql.GTE || n.Op == influxql.EQREGEX || n.Op == influxql.NEQREGEX {
+		switch n.Op {
+		case influxql.EQ, influxql.LT, influxql.LTE, influxql.GT, influxql.GTE, influxql.EQREGEX, influxql.NEQREGEX, influxql.NEQ:
```

Comment:
The fix is here -- add influxql.NEQ. I also changed to the code to use `switch` -- I think it's easier to read now.